File: src/main/java/com/bookmngsys/entity/Reader.java
package com.bookmngsys.entity;

import lombok.Data;

import javax.persistence.*;
import java.util.Date;

/**
 * @author Newway
 */
@Data
@Entity
public class Reader {
    /**
     * 读者id
     */
    @Id
    private int id;
    /**
     * 读者姓名
     */
    private String name;
    /**
     * 读者年龄
     */
    private int age;
    /**
     * 读者性别
     */
    private int sex;
    /**
     * 读者电话
     */
    private String phone;

    private Date regdate;

    @OneToOne
    @JoinColumn(name = "id")
    private User user;

    @ManyToOne(cascade = CascadeType.PERSIST)
    @JoinColumn(name = "level")
    private ReaderLevel level;
}

File: front-end/src/domain/reader.ts

export interface Reader {
    id:      number;
    name:    string;
    age:     number;
    sex:     number;
    phone:   string;
    regdate: number;
    user:    User;
    level:   Level;
}

export interface Level {
    id:       number;
    maxBrNum: number;
    maxBrDay: number;
}

export interface User {
    id:       number;
    username: string;
    password: string;
    type:     number;
    status:   number;
}
